locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsUtil.java

```diff
@@ -354,30 +355,60 @@ public static String getFirstReachableAddress(ComputeServiceContext context, Nod
         return getFirstReachableAddress(node, Duration.FIVE_MINUTES);
     }
     
+    /**
+     * Uses {@link Networking#isReachablePredicate()} to determine reachability.
+     * @see #getReachableAddresses(NodeMetadata, Duration, Predicate)
+     */
     public static String getFirstReachableAddress(NodeMetadata node, Duration timeout) {
-        return getFirstReachableAddress(node, timeout, new Networking.IsReachablePredicate());
+        return getFirstReachableAddress(node, timeout, Networking.isReachablePredicate());
     }
 
+    /** @see #getReachableAddresses(NodeMetadata, Duration, Predicate) */
     public static String getFirstReachableAddress(NodeMetadata node, Duration timeout, Predicate<? super HostAndPort> socketTester) {
+        Iterable<HostAndPort> addresses = getReachableAddresses(node, timeout, socketTester);
+        HostAndPort address = Iterables.getFirst(addresses, null);
+        if (address != null) {
+            return address.getHostText();
+        } else {
+            throw new IllegalStateException("No reachable IPs for " + node + "; check whether the node is " +
+                    "reachable and whether it meets the requirements of the HostAndPort tester: " + socketTester);
+        }
+    }
+
+    /**
+     * @return The public and private addresses of node that are reachable.
+     * @see #getReachableAddresses(Iterable, Duration, Predicate)
+     */
+    public static Iterable<HostAndPort> getReachableAddresses(NodeMetadata node, Duration timeout, Predicate<? super HostAndPort> socketTester) {
         final int port = node.getLoginPort();
```

Comment:
I acknowledge that this is existing code, but shouldn't we get the port from the port forwarder? External ports could be different from the actual listening port on the machine.